cloudformation/templates/gridss_worker_instance/worker.yaml

```diff
@@ -0,0 +1,221 @@
+AWSTemplateFormatVersion: 2010-09-09
+Resources:
+  workerInstance:
+    Type: 'AWS::EC2::Instance'
+    Properties:
+      AvailabilityZone: ap-southeast-2a
+      IamInstanceProfile: !Ref workerInstanceProfile
+      ImageId: ami-0dc96254d5535925f
+      InstanceType: !Ref instanceTypeParameter
+      SecurityGroupIds:
+        - !Ref instanceSecurityGroup
+      SubnetId: !Ref instanceSubnet
+      LaunchTemplate:
+        LaunchTemplateId: !Ref workerInstanceLauchTemplate
+        Version: !GetAtt 
+          - workerInstanceLauchTemplate
+          - LatestVersionNumber
+      KeyName: alexis-dev
+      Tags:
+        - Key: Source
+          Value: CloudFormation
+        - Key: Creator
+          Value: !Ref userName
+        - Key: Env
+          Value: Test
+        - Key: Name
+          Value: Worker instance
+        - Key: Stack
+          Value: !Ref AWS::StackId
+        - Key: Project
+          Value: gridss-purple-linx
+      UserData:
+        Fn::Base64: !Sub |
+            #!/bin/bash -xe
+            # Set time/logs to melbourne time
+            ln -sf /usr/share/zoneinfo/Australia/Melbourne /etc/localtime
+            # create mount point directory
+            mkdir /mnt/xvdh
+            # create ext4 filesystem on new volume           
+            mkfs -t ext4 /dev/xvdh
+            # add an entry to fstab to mount volume during boot
+            echo "/dev/xvdh       /mnt/xvdh   ext4    rw,suid,dev,exec,auto,user,async,nofail 0       2" >> /etc/fstab
+            # mount the volume on current boot
+            mount -a
+            # create a user folder for both users on this directory
+            mkdir /mnt/xvdh/ssm-user
+            mkdir /mnt/xvdh/ec2-user
+            # change the owner so the user (via SSM) has access
+            chown -R ssm-user /mnt/xvdh/ssm-user
+            chown -R ec2-user /mnt/xvdh/ec2-user
+            # install dev tools for later use
+            yum update -y -q
+            yum groups mark install "Development Tools"
+            yum update -y -q 
+            yum groupinstall -y 'Development Tools' 
+            # install docker
+            amazon-linux-extras install docker
+            # Move docker to larger diskspace
+            mv /var/lib/docker /mnt/xvdh/docker
+            # Link mount to docker container
+            ln -s /mnt/xvdh/docker /var/lib/docker
+            # Add users to docker
+            usermod -a -G docker ssm-user
+            usermod -a -G docker ec2-user
+            service docker start
+            # Install s3fuse from amazon
+            amazon-linux-extras install epel -y
+            yum install s3fs-fuse -y
```

Comment:
Why is fuse needed here? Normally with plain `aws cli` you get better performance, reliability (MD5 checksums per multipart) and stability. With S3FS you are renouncing to all that :/

Reply:
Yep, left over from a previous experiment set seeing if I could mount an S3, turns out I can without too much stress but later learning taught me that it wasn't common practise, hence there's not much documentation out there for it. I'll remove it from the bootstrap